Assets/MRTK/Providers/LeapMotion/Editor/ConfigurationChecker/LeapMotionConfigurationChecker.cs

```diff
@@ -0,0 +1,263 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.﻿
+
+using Microsoft.MixedReality.Toolkit.Utilities.Editor;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.LeapMotion
+{
+    /// <summary>
+    /// Class that checks if the Leap Motion Core assets are present and configures the project if they are.
+    /// </summary>
+    [InitializeOnLoad]
+    static class LeapMotionConfigurationChecker
+    {
+        private const string FileName = "README_BEFORE_UPDATING.txt";
+        private static readonly string[] definitions = { "LEAPMOTIONCORE_PRESENT" };
+        private static bool isLeapInProject = false;
+
+        static LeapMotionConfigurationChecker()
+        {
+            // Check if leap core is in the project
+            isLeapInProject = ReconcileLeapMotionDefine();
+
+            if (isLeapInProject)
+            {
+                RemoveTestingFolders();
+                AddAndUpdateAsmDefs();
+                AddLeapEditorAsmDefs();
+            }
+        }
+
+        /// <summary>
+        /// Ensures that the appropriate symbolic constant is defined based on the presence of the Leap Motion Core Assets.
+        /// </summary>
+        /// <returns>True if the define was added, false otherwise.</returns>
```

Comment:
Is this supposed to return true ONLY if the define was added by the function? i.e. if the defines were already there, then this technically no-op, but it would still return true. In general the language and specifics of return behavior should be super crisp

Reply:
This returns true when the define is added and if it was already added.  Added a more specific comment.